File: 1015_smallest_integer_divisible_by_k/smallest.java
class Solution {
    // Time: O(n)
    // Space: O(1)
    public int smallestRepunitDivByK(int K) {
        int remainder = 0;
        for (int res = 1; res <= K; res++) {
            remainder = (remainder * 10 + 1) % K;
            if (remainder == 0) {
                return res;
            }
        }
        return -1;
    }
}

File: 1015_smallest_integer_divisible_by_k/smallest.py
def smallestRepunitDivByK(K: int) -> int:
    """
    Time: O(n)
    Space: O(1)
    """
    remainder = 0
    for res in range(1, K + 1):
        remainder = (remainder * 10 + 1) % K
        if remainder == 0:
            return res
    return -1